slackbot/dispatcher.py

```diff
@@ -100,18 +107,17 @@ def loop(self):
 
     def _default_reply(self, msg):
         try:
-            from slackbot_settings import default_reply
+            from slackbot.settings import default_reply
+            default_reply = to_utf8(default_reply)
         except ImportError:
             default_reply = [
                 u'Bad command "{}", You can ask me one of the following questions:\n'.format(msg['text']),
             ]
-            default_reply += [u'    • `{0}` {1}'.format(p.pattern, v.__doc__ or "")
+            default_reply += ['    • `{0}` {1}'.format(p[1]
+                                                       , v.__doc__ or "")
                               for p, v in six.iteritems(self._plugins.commands['respond_to'])]
-            # pylint: disable=redefined-variable-type
-            default_reply = u'\n'.join(default_reply)
-
-        m = Message(self._client, msg)
-        m.reply(default_reply)
+            default_reply = '\n'.join(to_utf8(default_reply))
+        self._client.rtm_send_message(msg['channel'], default_reply)
```

Comment:
not clear what the changes to this section are actually doing, can you elaborate?